### DISCUSSION TITLE
Issue with setting up on termux on android 11

### DISCUSSION THREAD

**Discussion by jimkoutso2008** [ftde0/yt2009#193] [General]
All of the steps where fine without a problem, but when I try to run `node backend.js`, it errors out saying it can't access 0.0.0.0. Even though I have set up a proper local IP through the setup.
![Screenshot_20231227-015217338](https://github.com/ftde0/yt2009/assets/82086687/35cd4cf1-423c-438a-b99b-8ef32e540c01)

**Comment by EpicSentry** [comment]
You can't have anything listening on ports under 1024 without root. Are you running this on a rooted phone?

**Comment by jimkoutso2008** [comment]
Yes it's running on a rooted phone with magisk.
Also tried porting to :2009 (pun intended) and now it works

**Comment by breakgimme** [comment]
> Yes it's running on a rooted phone with magisk.

your phone is rooted, but you're not running termux as root
if you want to run yt2009 on port 80 you can install tsu and launch yt2009 with it

**Comment by jackhacksren** [comment]
> All of the steps where fine without a problem, but when I try to run `node backend.js`, it errors out saying it can't access 0.0.0.0. Even though I have set up a proper local IP through the setup. ![Screenshot_20231227-015217338](https://private-user-images.githubusercontent.com/82086687/292933008-35cd4cf1-423c-438a-b99b-8ef32e540c01.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3MDA2ODUsIm5iZiI6MTcwMzcwMDM4NSwicGF0aCI6Ii84MjA4NjY4Ny8yOTI5MzMwMDgtMzVjZDRjZjEtNDIzYy00MzhhLWI5OWItOGVmMzJlNTQwYzAxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDE4MDYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNjhmMDFjYjE1ZTY0ZDFmOTE1NDkxYjJlZTI3N2I2OTA4M2I2MzAzYTY0NTVmZGVhZWM1YWI1ZDg0MWU1MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5a_4H0dgrLPFKSADLkRwI0Jv2_I6NbfHcsoup1Kmrrc)

now how do you get it setup cuz i need module realine sync and npm install doesnt work

**Comment by aldairbotherotaku** [comment]
> > All of the steps where fine without a problem, but when I try to run `node backend.js`, it errors out saying it can't access 0.0.0.0. Even though I have set up a proper local IP through the setup. ![Screenshot_20231227-015217338](https://private-user-images.githubusercontent.com/82086687/292933008-35cd4cf1-423c-438a-b99b-8ef32e540c01.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM3MDA2ODUsIm5iZiI6MTcwMzcwMDM4NSwicGF0aCI6Ii84MjA4NjY4Ny8yOTI5MzMwMDgtMzVjZDRjZjEtNDIzYy00MzhhLWI5OWItOGVmMzJlNTQwYzAxLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI3VDE4MDYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNjhmMDFjYjE1ZTY0ZDFmOTE1NDkxYjJlZTI3N2I2OTA4M2I2MzAzYTY0NTVmZGVhZWM1YWI1ZDg0MWU1MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5a_4H0dgrLPFKSADLkRwI0Jv2_I6NbfHcsoup1Kmrrc)
> 
> now how do you get it setup cuz i need module realine sync and npm install doesnt work

use a chroot then, there is no real need of root. proot should fix this problem

**Comment by aldairbotherotaku** [comment]
Use https://github.com/TermuxArch/TermuxArch or see youtube.com/watch?v=0avuI2g4uOM for learn how to use chroot on termux

**Comment by CubeFun** [comment]
You need to have root, if you have it type "pkg install tsu" then type tsu and problem solved
